model/validation.go

```diff
@@ -669,3 +661,89 @@ func ValidateDestinationPolicy(msg proto.Message) error {
 
 	return errs
 }
+
+func validateProxyAddress(hostAddr string) error {
```

Comment:
The current code doesn't validate the host part either. Maybe something like the following?

```golang
host, port, err := net.SplitHostPort(hostAddr)
if err != { return err }

if _, err := net.ResolveIPAddr("ip6", host); err == nil {
    return fmt.Errorf("ipv6 not supported: %v", err)
}
// port validation, etc.
```